nats.go

```diff
@@ -1634,6 +1652,18 @@ func (nc *Conn) createConn() (err error) {
 		return ErrNoServers
 	}
 
+	// If we have a reference to an in-process server then establish a
+	// connection using that.
+	if nc.Opts.InProcessServer != nil {
+		conn, err := nc.Opts.InProcessServer.InProcessConn()
```

Comment:
I am bad at naming so I am not going to complain, but you will agree that there is a lot of stuttering here ;-) (InProcessServer.InProcessConn()).

Reply:
Indeed, if anyone has any better naming ideas I’m open to them!

Reply:
Maybe?
```
type InProcessConnProvider interface {
	Conn() (net.Conn, error)
}
conn, err := nc.Opts.InProcessServer.Conn()
```

Reply:
Agree with Jaime (@variadico). Change the interface and call here.